File: corpus/typescript/fixture_18.ts
type CartItem18 = { price: number; qty?: number };
export function calculateDiscount18(cartItems: CartItem18[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 180 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook18 {
  lookupPrice(sku: string): number { return sku.length + 18; }
}

File: corpus/rust/fixture_18.rs
pub struct PriceBook18 {
    pub multiplier: f64,
}

pub fn calculate_discount_18(cart_items: &[(f64, f64)], tax_rate: f64) -> f64 {
    let total_amount: f64 = cart_items.iter().map(|(price, qty)| price * qty).sum();
    if total_amount > 180.0 { total_amount * (1.0 + tax_rate) } else { total_amount }
}
